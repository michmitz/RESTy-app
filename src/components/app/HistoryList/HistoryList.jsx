import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from '../HistoryItem/HistoryItem';

const HistoryList = ({ history }) => {
  const historylist = history.map(historyitem => (
    <li key={`${historyitem.url}+${historyitem.method}`}>
        <HistoryItem {...historyitem} />
    </li>
  ));

  return (
    <ul data-testid="articles-list">
      {historylist}
    </ul>
  );
};

HistoryList.propTypes = {
  history: PropTypes.arrayOf(PropTypes.shape({
    url: PropTypes.string.isRequired,
    method: PropTypes.string.isRequired
  })).isRequired
};

export default HistoryList;