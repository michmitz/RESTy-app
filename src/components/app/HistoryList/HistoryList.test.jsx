import React from 'react';
import { render, cleanup } from '@testing-library/react';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  afterEach(() => cleanup());
  it('renders History List', () => {
    const { asFragment } = render(<HistoryList
      history={[
        {
          url: "test-url.com",
          method: "GET"
        }
      ]}
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});