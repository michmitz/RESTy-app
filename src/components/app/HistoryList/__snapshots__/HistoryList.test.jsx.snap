// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders History List 1`] = `
<DocumentFragment>
  <ul
    data-testid="history-list"
  >
    <li>
      <div
        class="history-item"
      >
        <p>
          GET
        </p>
        <p>
          test-url.com
        </p>
      </div>
    </li>
  </ul>
</DocumentFragment>
`;
